tools/commentMonitor/main.go

```diff
@@ -14,219 +14,217 @@
+	}, nil
+}
+
+func newGithubClient(ctx context.Context, owner, repo string, pr int) (*githubClient, error) {
+	ghToken := os.Getenv("GITHUB_TOKEN")
+	ts := oauth2.StaticTokenSource(&oauth2.Token{AccessToken: ghToken})
+	tc := oauth2.NewClient(ctx, ts)
+	return &githubClient{
+		clt:   github.NewClient(tc),
+		owner: owner,
+		repo:  repo,
+		pr:    pr,
+	}, nil
+}
+
+func (c commentMonitorConfig) webhookExtract(w http.ResponseWriter, r *http.Request) {
+	defer r.Body.Close()
+	payload, err := ioutil.ReadAll(r.Body)
 	if err != nil {
-		log.Fatalln(err)
+		http.Error(w, "unable to read webhook body", http.StatusBadRequest)
+	}
+
+	// TODO: setup webhook security with a webhook secret.
+
+	// Setup commentMonitor client.
+	cmClient := commentMonitorClient{
+		allArgs:  make(map[string]string),
+		eventMap: c.eventMap,
 	}
 
-	// Parsing event.json.
-	event, err := github.ParseWebHook("issue_comment", data)
+	// Parse webhook event.
+	event, err := github.ParseWebHook(github.WebHookType(r), payload)
 	if err != nil {
-		log.Fatalln(err)
+		http.Error(w, "unable to parse webhook", http.StatusBadRequest)
 	}
 
 	switch e := event.(type) {
 	case *github.IssueCommentEvent:
 
-		owner := *e.GetRepo().Owner.Login
-		repo := *e.GetRepo().Name
-		pr := *e.GetIssue().Number
-		author := *e.Sender.Login
-		authorAssociation := *e.GetComment().AuthorAssociation
-		commentBody := *e.GetComment().Body
-
-		// Setup commentMonitorClient.
+		// Setup github client.
 		ctx := context.Background()
-		cmClient.ghClient = newGithubClient(ctx, owner, repo, pr)
-
-		// Validate comment if regexString provided.
-		if cmClient.regexString != "" {
-			cmClient.regex = regexp.MustCompile(cmClient.regexString)
-			if !cmClient.regex.MatchString(commentBody) {
-				log.Fatalf("matching command not found. comment validation failed")
-			}
-			log.Println("comment validation successful")
+		cmClient.ghClient, err = newGithubClientForIssueComments(ctx, e)
+		if err != nil {
+			log.Println(err)
+			http.Error(w, "could not create GitHub client", http.StatusBadRequest)
+			return
 		}
 
-		// Verify if user is allowed to perform activity.
-		if !cmClient.verifyUserDisabled {
-			var allowed bool
-			allowedAssociations := []string{"COLLABORATOR", "MEMBER", "OWNER"}
-			for _, a := range allowedAssociations {
-				if a == authorAssociation {
-					allowed = true
-				}
-			}
-			if !allowed {
-				log.Printf("author is not a member or collaborator")
-				b := fmt.Sprintf("@%s is not a org member nor a collaborator and cannot execute benchmarks.", author)
-				if err := cmClient.ghClient.postComment(ctx, b); err != nil {
-					log.Fatalf("%v : couldn't post comment", err)
-				}
-				os.Exit(1)
-			}
-			log.Println("author is a member or collaborator")
+		// Validate regex.
+		err := cmClient.validateRegex()
+		if err != nil {
+			//log.Println(err) // Don't log on failure.
+			http.Error(w, "comment validation failed", http.StatusBadRequest)
```

Comment:
with most api's when you send a non 200 ok reply the same request is repeated which is not what we want here. Best to just return 200

I think `validateRegex()` should use bool as the return statment